src/axolotl/utils/gradient_checkpointing/unsloth.py

```diff
@@ -13,8 +13,19 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+from importlib.metadata import version
+
 import torch
 
+torch_version = version("torch")
+
+if torch_version < "2.4.0":
```

Comment:
`torch.version` is brittle, let's use some other version parsing check

Reply:
fixed